libs/pages/logs/environment/src/lib/ui/sidebar-pipeline/sidebar-pipeline.tsx

```diff
@@ -13,16 +14,22 @@ export function SidebarPipeline({ services, versionId, serviceId, statusStages }
   return (
     <div className="p-5">
       <p className="text-neutral-50 text-xs mb-4 font-medium">Pipeline</p>
-      {statusStages?.map((currentStage: DeploymentStageWithServicesStatuses, index: number) => (
-        <SidebarPipelineItem
-          key={index}
-          index={index}
-          versionId={versionId}
-          serviceId={serviceId}
-          currentStage={currentStage}
-          services={services}
-        />
-      ))}
+      {!statusStages ? (
+        <div className="flex justify-center">
+          <LoaderSpinner className="w-6 h-6" theme="dark" />
+        </div>
+      ) : (
+        statusStages.map((currentStage: DeploymentStageWithServicesStatuses, index: number) => (
+          <SidebarPipelineItem
+            key={index}
```

Comment:
`key` should not be index, use unique identifier instead like ```key={`${serviceId}${versionId}`}```

Reply:
I agree with this rule, but here we can't do that, we don't have a unique id on the `currentStage` and the `serviceId` +  `versionId` is the same for each Item

Reply:
done